sdk/tables/azure-data-tables/azure/data/tables/_error.py

```diff
@@ -195,6 +195,24 @@ def _decode_error(response, error_message=None, error_type=None, **kwargs):  # p
 
 
 def _reraise_error(decoded_error):
+    error_code = decoded_error.error_code
+    message = decoded_error.message
+    invalid_input = "The number of keys specified in the URI does not match number of key properties for the resource"
+    invalid_query_parameter_value = "Value for one of the query parameters specified in the request URI is invalid"
+    not_implemented = "The requested operation is not implemented on the specified resource"
+    properties_need_value = "The values are not specified for all properties in the entity"
+    table_does_not_exist = "The table specified does not exist"
+    
+    if (error_code == "InvalidQueryParameterValue" and invalid_query_parameter_value in message or
+        error_code == "InvalidInput" and invalid_input in message or
+        error_code == "NotImplemented" and not_implemented in message or
+        error_code == "PropertiesNeedValue" and properties_need_value in message or
+        error_code =="TableNotFound" and table_does_not_exist in message
+    ):
```

Comment:
I'm not sure we should put all this logic together in one place.
I think it would probably be more resilient to put the handling in the individual methods where the errors are raised.